crypto/x509/v3_addr.c

```diff
@@ -285,7 +285,7 @@ static int i2r_IPAddrBlocks(const X509V3_EXT_METHOD *method,
 static int IPAddressOrRange_cmp(const IPAddressOrRange *a,
                                 const IPAddressOrRange *b, const int length)
 {
-    unsigned char addr_a[ADDR_RAW_BUF_LEN], addr_b[ADDR_RAW_BUF_LEN];
+    unsigned char addr_a[ADDR_RAW_BUF_LEN] = {0}, addr_b[ADDR_RAW_BUF_LEN] = {0};
```

Comment:
Why is this an issue? The first parameter of `addr_expand()` is an output parameter, so those calls are okay

If you're concerned about `a->type` or `b->type` not being handled by the `switch()` statements, this change will mean that any such inputs will return `0`, which is probably not what is desired (the comment above the function says " we just arbitrarily declare that if given invalid inputs this function returns -1")

Wouldn't that case be better served by `default` statements in the `switch`es?

Reply:
As there is no default case for a->type or b->type in the switch() statements, If the code does not fail to any case then memcmp() will be done on garbage data. This will encounter the incorrect return of the function. So I initialize the char array.

Reply:
So why not add `default`s to the `switch`es to `return -1`?

Reply:
I agree.
Thanks for your valuable input.

Reply:
Still it is important to initialize **addr_a** and **addr_b**.

In addr_expand() function,
 memcpy(addr, bs->data, bs->length);

Here, it is possible that bs->length may be less than memory size (ADDR_RAW_BUF_LEN) pointed by addr. 
In this case, addr_a or addr_b may have garbage data.

Reply:
> In addr_expand() ... it is possible that bs->length may be less than memory size (ADDR_RAW_BUF_LEN) pointed by addr

Yes, in which case it executes

```
memset(addr + bs->length, fill, length - bs->length);
```

to fill the remainder of the output array.

> In this case, addr_a or addr_b may have garbage data

So this can't happen.